examples/step-20/step-20.cc

```diff
@@ -559,206 +566,86 @@ namespace Step20
+  // will therefore not discuss the rationale for our approach here any
+  // more, but rather only comment on some remaining implementational
+  // aspects.
 
+  // @sect4{MixedLaplace::solve}
 
-  void SchurComplement::vmult(Vector<double> &      dst,
-                              const Vector<double> &src) const
+  // As already outlined in the introduction, the solve function consists
+  // essentially of two steps. First, we have to form the first equation
+  // involving the Schur complement and solve for the pressure (component 1
+  // of the solution). Then, we can reconstruct the velocities from the
+  // second equation (component 0 of the solution).
+  template <int dim>
+  void MixedLaplaceProblem<dim>::solve()
   {
-    system_matrix->block(0, 1).vmult(tmp1, src);
-    m_inverse->vmult(tmp2, tmp1);
-    system_matrix->block(1, 0).vmult(dst, tmp2);
-  }
+    // As a first step we declare references to all block components of the
+    // matrix, the right hand side and the solution vector that we will
+    // need.
+    const auto &M = system_matrix.block(0, 0);
+    const auto &B = system_matrix.block(0, 1);
 
+    const auto &F = system_rhs.block(0);
+    const auto &G = system_rhs.block(1);
 
-  // @sect4{The <code>ApproximateSchurComplement</code> class}
+    auto &U = solution.block(0);
+    auto &P = solution.block(1);
 
-  // The third component of our solver and preconditioner system is the class
-  // that approximates the Schur complement with the method described in the
-  // introduction. We will use this class to build a preconditioner for our
-  // system matrix.
-  class ApproximateSchurComplement : public Subscriptor
-  {
-  public:
-    ApproximateSchurComplement(const BlockSparseMatrix<double> &A);
+    // Then, we will create corresponding LinearOperator objects and create
+    // the <code>op_M_inv</code> operator:
+    const auto op_M = linear_operator(M);
+    const auto op_B = linear_operator(B);
 
-    void vmult(Vector<double> &dst, const Vector<double> &src) const;
+    ReductionControl     reduction_control_M(2000, 1.0e-18, 1.0e-10);
+    SolverCG<>           solver_M(reduction_control_M);
+    PreconditionJacobi<> preconditioner_M;
 
-  private:
-    const SmartPointer<const BlockSparseMatrix<double>> system_matrix;
+    preconditioner_M.initialize(M);
 
-    mutable Vector<double> tmp1, tmp2;
-  };
+    const auto op_M_inv = inverse_operator(op_M, solver_M, preconditioner_M);
 
+    // This puts us in the position to be able to declare the Schur
+    // complement <code>op_S</code> and the approximate Schur complement
+    // <code>op_aS</code>:
+    const auto op_S = transpose_operator(op_B) * op_M_inv * op_B;
+    const auto op_aS =
+      transpose_operator(op_B) * linear_operator(preconditioner_M) * op_B;
 
+    // We now create a preconditioner out of <code>op_aS</code> that
+    // applies a few number of CG iterations (until a very modest relative
```

Comment:
... applies a few *CG iterations*...

I am not a native speaker so it is difficult to dismiss this clause, but it sounds a bit strange, so I would skip "number of"

Reply:
Correct. Removed the "number of".